src/audio/kpb.c

```diff
@@ -969,34 +1093,39 @@ static int kpb_copy(struct comp_dev *dev)
 			ret = PPL_STATUS_PATH_STOP;
 			break;
 		}
+		if (kpb->mic_sel_cnt == 0) {
+			kpb_copy_samples(sink_c, source_c, copy_bytes, sample_width, channels);
+			avail_bytes = audio_stream_get_avail_bytes(&source_c->stream);
 
-		kpb_copy_samples(sink_c, source_c, copy_bytes, sample_width, channels);
+			/* Buffer source data internally in history buffer for future
+			 * use by clients.
+			 */
+			if (avail_bytes <= kpb->hd.buffer_size) {
+				ret = kpb_buffer_data(dev, source_c, copy_bytes);
```

Comment:
now the data is buffered in history-buffer only in non-micselect case
I think if/else should only select between kpb_copy_samples() / kpb_micselect_copy() and leave history-buffer part unchanged.